### PR TITLE
Create behavior reference for Lifecycle for Edge Network extension

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR creates a new Lifecycle for Edge Network reference doc that explains in depth the different lifecycle event flows and pitfalls commonly seen when implementing Lifecycle for Edge Network.

It is similar to the Lifecycle behavior reference created in #568, but for Lifecycle for Edge Network's specific behavior.
* Similar diagram intents etc as explained in linked PR

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/edge/lifecycle-for-edge-network/behavior-reference.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+    classDef regularBox fill:#009c3b,stroke:#009c3b,color:#fff;
+    classDef incorrectBox fill:#EB1000,stroke:#EB1000,color:#fff;
+    classDef transparentSubgraph fill:transparent,stroke:#7a7a7a;
+
+    class B,C,F regularBox;
+    class E incorrectBox;
+ -->
+
+## Expected Lifecycle scenarios
```
Comment (kevinlind) [inline src/pages/edge/lifecycle-for-edge-network/behavior-reference.md:117]: I'd suggest to swap the ordering of this section with the previous "Consecutive Lifecycle API Calls" so the expected scenarios are listed first.

File: src/pages/edge/lifecycle-for-edge-network/behavior-reference.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+The following Lifecycle workflows show examples of the expected ordering of Lifecycle events.
+
+### App launch and close
+
+This scenario shows the standard Lifecycle workflow, with expected Lifecycle API calls.
+
+![](./assets/index/lifecycle-app-launch-and-close.svg)
```
Comment (kevinlind) [inline src/pages/edge/lifecycle-for-edge-network/behavior-reference.md:125]: In this image `lifecycle-app-launch-and-close.svg`, can you include "(Close type close)" under "Application Close (Background)", similar to how you have "(Close type unknown)" in the `lifecycle-crash.svg` image?

File: src/pages/edge/lifecycle-for-edge-network/behavior-reference.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+
+For full event details, please read the [Lifecycle event reference](../../home/base/mobile-core/lifecycle/event-reference.md).
+
+#### Data payloads
+
+The Lifecycle for Edge Network event payloads are defined by the Platform Mobile Lifecycle Details XDM field group, and include information about the application, device, and environment when the event occurred. Please refer to [Lifecycle for Edge Network data](metrics.md) for the data included with these events.
+
+As a consequence of this shift in not relying on a session timeout window to consider new sessions, the potential Lifecycle states are greatly simplified:
+
+### Consecutive Lifecycle API calls
```
Comment (cacheung) [inline src/pages/edge/lifecycle-for-edge-network/behavior-reference.md:43]: The heading size here is confusing.  
Is Consecutive Lifecycle API calls under General behavior?

Why it is 
##
####
###?

For 
"As a consequence of this shift in not relying on a session timeout window to consider new sessions, the potential Lifecycle states are greatly simplified:" 
Is it part of Data payloads?  I can't tell by reading with the headings.

File: src/pages/edge/lifecycle-for-edge-network/behavior-reference.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+    classDef incorrectBox fill:#EB1000,stroke:#EB1000,color:#fff;
+    classDef transparentSubgraph fill:transparent,stroke:#7a7a7a;
+
+    class A,D regularPill;
+    class B,C,E,F regularBox;
+    class graph1,graph2 transparentSubgraph;
+-->
+
+### On app crash or force close
+
```
Comment (cacheung) [inline src/pages/edge/lifecycle-for-edge-network/behavior-reference.md:162]: Is this scenario same as the above Consecutive 1a. lifecycleStart, 2a lifecycleStart?
Or the consecutive lifecycleStart in the above has to be happened in the same session without a forced closed?

**Comment by cacheung** [comment]
I wonder if there is a way to identify between the lifecycleStart or Pause was dispatched by result of an API and the lifecycleStart or Pause was dispatched by SDK due to other reasons. (may be different text color?)

**Review by calebk1 — APPROVED** [review]
LGTM!

**Review by kevinlind — APPROVED** [review]

**Review by cacheung — APPROVED** [review]

**Pull request merged**